ex4B/ex4.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Write a function that convers a string into number; return null if the string
+ * doesn't contain a valid number.
+ *
+ * For example:
+ * toNumber("71") === 71
+ * toNumber("a7") === null
+ */
+function toNumber(str) {
+
+	var tmpStr = '', i = 0, dotRepeat = null;
```

Comment:
Also, `dotRepeat` sounds like there's an option to allow several dots, which is not the case. How about renaming the variable to `dotAppeared` (`false` at first, `true` after finding a dot -- same as you have now) or `allowDot` (`true` at first, `false` after finding a dot -- inverse of what you have now).